docs/content/v2025.1/explore/transactions/transactional-ddl.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Transactional DDL
+headerTitle: Transactional DDL
+linkTitle: Transactional DDL
+description: Learn how YugabyteDB handles DDLs in a transaction
+headcontent: Learn how YugabyteDB handles DDLs in a transaction
+tags:
+   feature: tech-preview
+menu:
+  v2025.1:
+    identifier: transactional-ddl
+    parent: explore-transactions
+    weight: 300
+type: docs
+---
+
+Data Definition Language (DDL) statements modify the structure of your data. PostgreSQL allows executing such statements within a transactional block and supports rolling them back as part of a transaction rollback. For two transactions undergoing conflicting DDLs, it also provides isolation guarantees by taking appropriate object locks during DDL processing.
+
+YugabyteDB's transactional DDL provides similar guarantees for rolling back DDL operations done inside a transaction block. However, because object locks are still under active development, isolation guarantees are weaker than in PostgreSQL. See [Limitations](#limitations) for more information.
+
+## Enable transactional DDL
+
+{{<tags/feature/tp idea="1677">}} Support for transactional DDL is disabled by default, and to enable the feature, set the [yb-tserver](../../../reference/configuration/yb-tserver/) flag `yb_ddl_transaction_block_enabled` to true.
+
+Because `yb_ddl_transaction_block_enabled` is a preview flag, to use it, add the flag to the [allowed_preview_flags_csv](../../../reference/configuration/yb-tserver/#allowed-preview-flags-csv) list (that is, `allowed_preview_flags_csv=yb_ddl_transaction_block_enabled`).
```

Comment:
```suggestion
Because `ysql_yb_ddl_transaction_block_enabled` is a preview flag, to use it, add the flag to the [allowed_preview_flags_csv](../../../reference/configuration/yb-tserver/#allowed-preview-flags-csv) list (that is, `allowed_preview_flags_csv=ysql_yb_ddl_transaction_block_enabled`).
```